src/N98/Magento/Command/Github/PullRequestCommand.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace N98\Magento\Command\Github;
+
+use N98\Magento\Command\AbstractMagentoCommand;
+use N98\Util\OperatingSystem;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use WpOrg\Requests\Requests;
+
+class PullRequestCommand extends AbstractMagentoCommand
+{
+    private string $diffContent = '';
+
+    private string $repository = '';
+
+    protected function configure()
+    {
+        $this->setName('github:pr')
+            ->addArgument(
+                'number',
+                InputArgument::REQUIRED,
+                'Pull Request Number'
+            )
+            ->addOption('repository', 'r', InputOption::VALUE_OPTIONAL, 'Repository to fetch from', 'magento/magento2')
+            ->addOption('mage-os', null, InputOption::VALUE_NONE, 'Shortcut option to use the mage-os/mageos-magento2 repository.')
+            ->addOption('patch', 'd', InputOption::VALUE_NONE, 'Download patch and prepare it for applying')
+            ->addOption('diff', null, InputOption::VALUE_NONE, 'Raw diff download')
+            ->addOption('json', null, InputOption::VALUE_NONE, 'Show pull request data as json')
+            ->setDescription('Download patch from github merge request <comment>(experimental)</comment>');
+    }
```

Comment:
Would be nice to have an option to specify to which directory it should write the patch file(s)